multicluster/cmd/multicluster-controller/controller.go

```diff
@@ -88,33 +86,37 @@ func createClients(kubeConfig *rest.Config) (
 	return client, aggregatorClient, apiExtensionClient, nil
 }
 
-func getCaConfig(controllerNs string) *certificate.CAConfig {
+func getCaConfig() *certificate.CAConfig {
 	return &certificate.CAConfig{
 		CAConfigMapName: configMapName,
 		// the key pair name has to be "tls" https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/manager/manager.go#L221
-		PairName:           "tls",
-		CertDir:            certDir,
-		ServiceName:        serviceName,
-		SelfSignedCertDir:  selfSignedCertDir,
-		MutationWebhooks:   getMutationWebhooks(controllerNs),
-		ValidatingWebhooks: getValidationWebhooks(controllerNs),
-		CertReadyTimeout:   2 * time.Minute,
-		MaxRotateDuration:  time.Hour * (24 * 365),
+		PairName:                  "tls",
+		CertDir:                   certDir,
+		ServiceName:               serviceName,
+		SelfSignedCertDir:         selfSignedCertDir,
+		MutationWebhookSelector:   getMutationWebhooks(),
+		ValidatingWebhookSelector: getValidationWebhooks(),
+		CertReadyTimeout:          2 * time.Minute,
+		MaxRotateDuration:         time.Hour * (24 * 365),
 	}
 }
 
-func getValidationWebhooks(controllerNs string) []string {
-	if controllerNs != "" {
-		return []string{fmt.Sprintf(validationWebhooksNamePattern, controllerNs, "-")}
+func getValidationWebhooks() *metav1.LabelSelector {
+	return &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"app":       "antrea",
+			"served-by": "antrea-mc",
+		},
 	}
-	return []string{fmt.Sprintf(validationWebhooksNamePattern, "", "")}
 }
 
-func getMutationWebhooks(controllerNs string) []string {
-	if controllerNs != "" {
-		return []string{fmt.Sprintf(mutationWebhooksNamePattern, controllerNs, "-")}
+func getMutationWebhooks() *metav1.LabelSelector {
+	return &metav1.LabelSelector{
+		MatchLabels: map[string]string{
```

Comment:
I think you still need the `controllerNs` namespace as a filter or selector since user may run two antrea-mc-controllers in one cluster as leader and member, they are deployed in different Namespaces.

Reply:
updated.